soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -3721,8 +4015,14 @@ void CreateGetItemMessages(std::vector<GetItemMessage> messageEntries) {
 void CreateScrubMessages() {
     CustomMessageManager* customMessageManager = CustomMessageManager::Instance;
     customMessageManager->AddCustomMessageTable(Randomizer::scrubMessageTableID);
-    const std::vector<u8> prices = { 10, 40 };
-    for (u8 price : prices) {
+    customMessageManager->CreateMessage(Randomizer::scrubMessageTableID, 0,
+        { TEXTBOX_TYPE_BLACK, TEXTBOX_POS_BOTTOM,
+            "\x12\x38\x82\All right! You win! In return for&sparing me, I will give you a&%gmysterious item%w!&Please, take it!\x07\x10\xA3",
+            "\x12\x38\x82\In Ordnung! Du gewinnst! Im Austausch&dafür, dass du mich verschont hast,&werde ich dir einen %gmysteriösen&Gegenstand%w geben! Bitte nimm ihn!\x07\x10\xA3",
+            "\x12\x38\x82\D'accord! Vous avez gagné! En échange&de m'épargner, je vous donnerai un &%gobjet mystérieux%w! S'il vous plaît,&prenez-le!\x07\x10\xA3",
+        });
+
+    for (u32 price = 5; price <= 95; price += 5) {
```

Comment:
i get that you're making new messages for every price, and then pulling it out using
```cpp
messageEntry = Randomizer_GetScrubMessage((textId & ((1 << 8) - 1)));
```
it just feels a little strange to me

@leggettc18 if you have any thoughts on this use of the custom message system i'd love to hear them

Reply:
Well at the moment the price is the only thing that's different between all of these messages, so it made sense to me to use the price as a textId. Since then there's been talk of adding an option for the text to reference the item they sell with differing levels of obscurity (pretty much exactly like hints do currently) which would require either saving the text to the save file (and saving the price along with it) or substituting that item text at runtime (at which point we might as well do that for the price as well). So maybe it should be changed in the future, but this might be fine for the first pass.

Reply:
> or substituting that item text at runtime

This is the pattern I'm following for shopsanity, and plan to eventually migrate scrubs to, I got it most of the way working with scrubs but ran into a snag that I didn't want to hold up scrubsanity for.

Reply:
yeah, i'm cool with it as is too. i think having the bitwise logic in the calls is what was throwing me for a loop when trying to read through it the first time

Reply:
> This is the pattern I'm following for shopsanity, and plan to eventually migrate scrubs to, I got it most of the way working with scrubs but ran into a snag that I didn't want to hold up scrubsanity for.

good deal! we can make an issue for that to look into after getting this merged

Reply:
I think changing the scrubs table to a merchants table would be a good idea. Shopsanity messages can also go in there, and when we do shuffle magic beans and shuffle merchants the changed dialog to those can also go in there.

Reply:
Going to resolve this comment, but added a comment here explaining the plans